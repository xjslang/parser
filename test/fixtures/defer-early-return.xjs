function testEarlyReturn() {
  console.log('function start')
  defer {
    console.log('defer executed despite early return')
  }
  
  if (true) {
    console.log('early return')
    return 'early'
  }
  
  console.log('this should not execute')
}

testEarlyReturn()
