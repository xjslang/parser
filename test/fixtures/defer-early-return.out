function start
early return
defer executed despite early return
